core/src/main/java/org/elasticsearch/index/mapper/core/StringFieldMapper.java

```diff
@@ -66,7 +66,8 @@
     private static final Set<String> SUPPORTED_PARAMETERS_FOR_AUTO_UPGRADE = new HashSet<>(Arrays.asList(
             "type",
             // most common parameters, for which the upgrade is straightforward
-            "index", "store", "doc_values", "omit_norms", "norms", "fields", "copy_to", "fielddata"));
```

Comment:
would we awesome to get a comment here what this set if for ie. why things are straight forward and what happens if it's not in the set and we see it on upgrade